packages/edit-site/src/components/navigation-sidebar/navigation-panel/menus/content-posts.js

```diff
@@ -33,22 +43,41 @@ export default function ContentPostsMenu() {
 				queryContext: { page: 1 },
 			},
 		} );
-	};
+	}, [ setPage ] );
 
 	return (
 		<NavigationMenu
 			menu={ MENU_CONTENT_POSTS }
 			title={ __( 'Posts' ) }
 			parentMenu={ MENU_ROOT }
+			hasSearch={ true }
+			onSearch={ onSearch }
+			search={ search }
 		>
-			{ showOnFront === 'posts' && (
-				<NavigationItem
-					item={ 'content-/' }
-					title={ __( 'All Posts' ) }
-					onClick={ onActivateFrontItem }
-				/>
+			{ search && <SearchResults items={ posts } search={ search } /> }
+
+			{ ! search && (
+				<>
+					{ showOnFront === 'posts' && (
+						<NavigationItem
+							item={ 'post-/' }
+							title={ __( 'All Posts' ) }
+							onClick={ onActivateFrontItem }
+						/>
+					) }
```

Comment:
`All Posts` is not included in the search results. Should we also include this?

Reply:
I think it's fine to omit it. Furthermore I'm not sure if we should be displaying it in posts menu anyway since it sounds more like page template to me. This should probably be reserved just for individual posts and not a collection of them.